### ISSUE TITLE
Work Item-Level Authorization

### ISSUE THREAD

**Issue by alexeykazakov** [openshiftio/openshift.io#340]
As a User, I want to be able to limit access to work items in my spaces for other osio users.

**Comment by michaelkleinhenz** [comment]
I am unsure if this is a topic for the next 5-6 sprints. This would imho require a lot of ui dialogs, possibly with user-created roles and associations of roles to rights. I think this is something for beyond J1.

**Comment by alexeykazakov** [comment]
To be clear. We plan to implement some backend for this and switch the current collaborator functionality to the new backend. But father integration with OSIO clients/components and UI will go to the next train. We can't build a massive new backend and adopt it across OSIO in the same train.

**Comment by qodfathr** [comment]
@alexeykazakov agreed. We'll need to write a bunch of new issues for the specific permissions available in the service, but first we need to build the permission system. :)

I'm going to remove `heather` from this; I'll start scoping some proposed permissions and look for contributions from the team.
